dot/state/epoch.go

```diff
@@ -202,7 +202,11 @@ func (s *EpochState) GetEpochForBlock(header *types.Header) (uint64, error) {
 	}
 
 	for _, d := range header.Digest.Types {
-		predigest, ok := d.Value().(types.PreRuntimeDigest)
+		digestValue, err := d.Value()
+		if err != nil {
+			continue
```

Comment:
Should we panic here? :thinking:

Reply:
I don't think so. This function iterates through all digests to check for the preruntime digest. I think technically this should be the first digest anyways, but possibly there is a case where there are other empty digests that we check before preruntime digest.

Even if not, the current logic is that if we don't find the preruntime digest we return error and I'm not sure why we would change that in this case.